scripts/service-update-scripts/dcos-commons-merge.sh

```diff
@@ -0,0 +1,200 @@
+        echo "Package $package requires a version ending in '-beta', but version was $version"
+        exit 1
+    fi
+else
+    branch=$STABLE_BRANCH
+    # Verify that the version doesn't end with '-beta'
+    if [[ "$version" == *-beta ]]; then
+        echo "Package $package requires a version that doesn't end in '-beta', but version was $version"
+        exit 1
+    fi
+fi
+
+# Go to dcos-docs-site root directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd ${SCRIPT_DIR}/../..
+pwd
+
+# Grab correct branch from repo
+rm -rf $REPO_NAME/
+git clone --depth 1 --branch $branch $REPO_URL
+
+# Omit any 'beta-' prefix from package name when getting dcos-commons path:
+framework=$(echo $package | sed 's/^beta-//g')
+input_framework_docs_dir=$REPO_NAME/frameworks/${framework}/docs
+
+output_package_dir=pages/services/${package}
+output_package_version_dir=${output_package_dir}/${version}
+if [ -d $output_package_version_dir ]; then
+   echo "Output directory already exists, remove first: $(pwd)/$output_package_version_dir"
+   exit 1
+fi
+
+# Before copying content, add 10 to the sort order of existing index files
+for i in $( ls ${output_package_dir}/*/index.md ); do
+    awk '/^menuWeight:/ {sub(/[[:digit:]]+$/,$NF+10)}1 {print}' $i > $i.tmp && mv $i.tmp $i
+done
+
+merge_list_file=scripts/service-update-scripts/merge-lists/${package}-merge-list.txt
+rm -f $merge_list_file
+
+###
+# Version-specific content
+
+echo "Copying service content"
+rm -rf ${output_package_version_dir}
+mkdir -p ${output_package_version_dir}
+for filepath in $(ls $input_framework_docs_dir/*.md); do
+    if [ $(basename $filepath) == "index.md" ]; then
+        # Copy index.md to root, don't create index/index.md...
+        cp -v $filepath $output_package_version_dir/
+    else
+        # Copy foo.md as foo/index.md
+        directory_name=$(basename $filepath | sed 's/\.md$//g')
+        mkdir -p $output_package_version_dir/$directory_name
+        cp -v $filepath $output_package_version_dir/$directory_name/index.md
+    fi
+    echo $filepath >> $merge_list_file
+done
+
+if [ -d $input_framework_docs_dir/img/ ]; then
+    echo "Copying service images"
+    mkdir -p ${output_package_version_dir}/img/
+    cp -v $input_framework_docs_dir/img/* ${output_package_version_dir}/img/
+    ls $input_framework_docs_dir/img/* >> $merge_list_file
+fi
+
+echo "Updating service model"
+# e.g.: cassandra.yml => services/beta-cassandra/data.yml
+output_yml_data_filepath=pages/services/${package}/data.yml
+cp -v $REPO_NAME/docs/pages/_data/services/${framework}.yml $output_yml_data_filepath
+
+###
+# Shared content
+
+echo "Updating template content"
+input_template_dir=$REPO_NAME/docs/pages/_includes/services
+output_template_dir=pages/services/include
+
+rm -rf $output_template_dir
+mkdir -p $output_template_dir
+for filepath in $(ls $input_template_dir/*.md); do
+    # e.g. pages/_includes/services/template.md => pages/services/include/template.tmpl
```

Comment:
^-- SC2045: Iterating over ls output is fragile. Use globs
https://github.com/koalaman/shellcheck/wiki/SC2045